packages/queries/src/linodes/interfaces.ts

```diff
@@ -240,13 +241,33 @@ export const useUpgradeToLinodeInterfacesMutation = (linodeId: number) => {
   return useMutation<UpgradeInterfaceData, APIError[], UpgradeInterfacePayload>(
     {
       mutationFn: (data) => upgradeToLinodeInterface(linodeId, data),
-      onSuccess() {
-        queryClient.invalidateQueries({
-          queryKey: linodeQueries.linode(linodeId).queryKey,
-        });
-        queryClient.invalidateQueries({
-          queryKey: linodeQueries.linode(linodeId)._ctx.configs.queryKey,
-        });
+      onSuccess(upgradeData) {
+        // only invalidate queries if this is an actual upgrade, not a dry run
+        if (upgradeData.dry_run === false) {
+          queryClient.invalidateQueries({
+            queryKey: linodeQueries.linode(linodeId).queryKey,
+          });
+          queryClient.invalidateQueries({
+            queryKey: linodeQueries.linode(linodeId)._ctx.configs.queryKey,
+          });
```

Comment:
Same thing here, invalidating `linodeQueries.linode(linodeId).queryKey` would include invalidating `linodeQueries.linode(linodeId)._ctx.configs.queryKey` because of how query keys work based on prefix

I'm not sure exactly what we want here


```suggestion
          queryClient.invalidateQueries({
            queryKey: linodeQueries.linode(linodeId).queryKey,
            exact: true,
          });
          queryClient.invalidateQueries({
            queryKey: linodeQueries.linode(linodeId)._ctx.configs.queryKey,
          });
```
would invalidate the Linode and the Linode's configs

```suggestion
          queryClient.invalidateQueries({
            queryKey: linodeQueries.linode(linodeId).queryKey,
          });
```
invalidates all of a Linode's queries (the Linode Itself, Configs, Interfaces, Disks, Volumes, Firewalls, ....)

Reply:
added! also added the subnet linode row thing - I think what I just committed should do the trick for there